internal/langserver/handlers/did_change_watched_files.go

```diff
@@ -0,0 +1,40 @@
+package handlers
+
+import (
+	"context"
+
+	lsctx "github.com/hashicorp/terraform-ls/internal/context"
+	ilsp "github.com/hashicorp/terraform-ls/internal/lsp"
+	"github.com/hashicorp/terraform-ls/internal/protocol"
+	lsp "github.com/hashicorp/terraform-ls/internal/protocol"
+)
+
+func DidChangeWatchedFiles(ctx context.Context, params lsp.DidChangeWatchedFilesParams) error {
+	fs, err := lsctx.DocumentStorage(ctx)
+	if err != nil {
+		return err
+	}
+
+	for _, change := range params.Changes {
+		fh := ilsp.FileHandlerFromDocumentURI(change.URI)
+		switch change.Type {
+		case protocol.Created:
+			err := fs.CreateAndOpenDocument(fh, fh.Filename(), []byte(""))
+			if err != nil {
+				return err
+			}
+
+		case protocol.Changed:
+			// fs.ChangeDocument(..)
+			// fs.GetDocument(..)
```

Comment:
I'm not exactly clear if it's just okay to load this "changed" document. Since there's no content or other meta for the document, I'm guessing this is the way to move forward? If yes, then should we check if it's part of a module and accordingly load the doc again?

Reply:
cc/ @radeksimko